tee-worker/litentry/pallets/identity-management/src/lib.rs

```diff
@@ -194,150 +147,62 @@ pub mod pallet {
 
 		#[pallet::call_index(1)]
 		#[pallet::weight(15_000_000)]
-		pub fn set_challenge_code(
+		pub fn link_identity(
 			origin: OriginFor<T>,
 			who: T::AccountId,
 			identity: Identity,
-			code: ChallengeCode,
+			parent_ss58_prefix: u16,
 		) -> DispatchResult {
 			T::ManageOrigin::ensure_origin(origin)?;
-			// we don't care if it has already associated with any challenge code
-			ChallengeCodes::<T>::insert(&who, &identity, code);
-			Self::deposit_event(Event::ChallengeCodeSet { who, identity, code });
-			Ok(())
-		}
 
-		#[pallet::call_index(2)]
-		#[pallet::weight(15_000_000)]
-		pub fn remove_challenge_code(
-			origin: OriginFor<T>,
-			who: T::AccountId,
-			identity: Identity,
-		) -> DispatchResult {
-			T::ManageOrigin::ensure_origin(origin)?;
 			ensure!(
-				ChallengeCodes::<T>::contains_key(&who, &identity),
-				Error::<T>::ChallengeCodeNotExist
+				!IDGraphs::<T>::contains_key(&who, &identity),
```

Comment:
Ensuring the identity cannot be linked to another account is out of the scope of this PR ?

Reply:
That's right, it belongs to EVM-sign-in feature or a separate issue if need.